crates/storage/src/module_cache.rs

```diff
@@ -151,32 +139,35 @@ impl InitiaModuleCache {
             Extension = BytesWithHash,
             Version = NoVersion,
         >,
-    ) -> VMResult<Option<Arc<ModuleCode<CompiledModule, Module, BytesWithHash, NoVersion>>>> {
+    ) -> VMResult<Option<ModuleWrapper>> {
         let mut module_cache = self.module_cache.lock();
         Ok(match module_cache.get(checksum) {
-            Some(code) => Some(code.clone()),
-            None => match builder.build(id)? {
-                Some(code) => {
-                    if code.extension().hash() != checksum {
-                        return Err(PartialVMError::new(
-                            StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR,
-                        )
-                        .with_message("Checksum mismatch".to_string())
-                        .finish(Location::Module(id.clone())));
+            Some(module_wrapper) => Some(module_wrapper.clone()),
+            None => {
+                let (build_result, allocated_size) = get_size(move || builder.build(id))?;
+                match build_result {
+                    Some(code) => {
+                        if code.extension().hash() != checksum {
+                            return Err(PartialVMError::new(
+                                StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR,
+                            )
+                            .with_message("Checksum mismatch".to_string())
+                            .finish(Location::Module(id.clone())));
+                        }
+
+                        let code_wrapper = ModuleWrapper::new(Arc::new(code), allocated_size);
+                        module_cache
+                            .put_with_weight(*checksum, code_wrapper.clone())
+                            .map_err(|_| {
```

Comment:
maybe we don't need to return error when cache put failed due to size limit?

Reply:
it occurs error when the size of the entry is larger than total cache capacity. so, once it occurs, it will not continue to enter the cache.

Reply:
so I mean, this cache size is node operator dependent. so someone can set this value small and if a huge module is published then some nodes may meet this cache error due to local config and someone not. to prevent this non-deterministic case, we may better to ignore this cache error and just return `code_wrapper`. 

Or better to check size limit first and not try to put huge module if the module size is bigger than total size.

Reply:
yeah right, good point.